### PR TITLE
Issue 118 & 83 (insist on maven and or cradle flags on project create & runs get supports active flag)

### PR DESCRIPTION
**Pull request by Wyvinar**
Error seems to have been made in commits. The first commit is supposed to go towards issue 83, cli supports --active flag.
Second commit is for the issue 118 project create should insist on either --gradle, --maven, or both flags.

### CODE REVIEW

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-cli-pr-130-72v5m for details. If you are unable to do so, please contact a member of the Galasa team.

File: pkg/runs/runsGet_test.go [hunk 1]
```diff
@@ -1204,10 +1226,100 @@ func TestRunsGetURLQueryWithInvalidResultSuppliedReturnsError(t *testing.T) {
...
+
+	apiServerUrl := server.URL
+	mockTimeService := utils.NewMockTimeService()
+
+	// When...
+	err := GetRuns(runName, age, requestor, result, shouldGetActive, outputFormat, mockTimeService, mockConsole, apiServerUrl)
+
+	// Then ...
+	assert.Nil(t, err)
+}
```
Comment (jadecarino) [inline pkg/runs/runsGet_test.go:1284]: Is there any way we can mock out a response in this unit test and return an active run?

File: pkg/runs/runsGet_test.go [hunk 2]
```diff
@@ -1204,10 +1226,100 @@ func TestRunsGetURLQueryWithInvalidResultSuppliedReturnsError(t *testing.T) {
...
+
+// 	apiServerUrl := server.URL
+// 	mockTimeService := utils.NewMockTimeService()
+
+// 	// When...
+// 	err := GetRuns(runName, age, requestor, result, shouldGetActive, outputFormat, mockTimeService, mockConsole, apiServerUrl)
+
+// 	// Then ...
+// 	assert.Nil(t, err)
+// }
```
Comment (jadecarino) [inline pkg/runs/runsGet_test.go:1325]: As above, why is this commented out? If for a reason then please can you add a comment above describing why it's commented, so if someone else comes across it they know. If it's not needed then please remove.

**Review by jadecarino — CHANGES_REQUESTED** [review]

**Comment by jadecarino** [comment]
Hi Holly, good work thanks for making these changes. Just a few comments to address but otherwise good. Also I know you mentioned there was a mistake but in future if you could try to keep separate PRs and branches for different work items so it's easier to review for the reviewer and smaller changes being merged in that would be great! :)

File: pkg/runs/runsGet_test.go (hunk 1 above)
Comment (Wyvinar) [reply pkg/runs/runsGet_test.go:1284]: Do you have any examples I could use as a guide?

File: pkg/runs/runsGet.go [hunk 3]
```diff
@@ -243,6 +254,9 @@ func GetRunsFromRestApi(
 		if resultParameter != "" {
 			apicall = apicall.Result(resultParameter)
 		}
+		// if shouldGetActive {
+		// 	apicall = apicall.Status(activeStatusNames)
+		// }
```
[the remark this answers is not in the document]
Comment (KirbyKatcher) [reply pkg/runs/runsGet.go]: This code was commented out because it needed the api changes to work. I have uncommented it and tested.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-cli-pr-130-65rbr for details. If you are unable to do so, please contact a member of the Galasa team.

File: pkg/runs/runsGet_test.go (hunk 1 above)
Comment (KirbyKatcher) [reply pkg/runs/runsGet_test.go:1284]: corrected the unit test

File: pkg/runs/runsGet_test.go (hunk 2 above)
Comment (KirbyKatcher) [reply pkg/runs/runsGet_test.go:1325]: refactored to check the status is passed correctly in the url query

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-cli-pr-130-qbj6l for details. If you are unable to do so, please contact a member of the Galasa team.

File: pkg/cmd/projectCreate.go [hunk 4]
```diff
@@ -158,41 +158,42 @@ func createProject(
...
-	if useGradle && !useMaven {
-		useMaven = false
-	} else {
-		useMaven = true
+	if !useGradle && !useMaven {
+		err = galasaErrors.NewGalasaError(galasaErrors.GALASA_ERROR_MAVEN_AND_OR_GRADLE_FLAG_MUST_BE_SET)
 	}
 
-	embeddedFileSystem := embedded.GetReadOnlyFileSystem()
+	if err == nil {
```
Comment (techcobweb) [inline pkg/cmd/projectCreate.go:165]: This could/should be an 'else' ?

**Review by techcobweb — APPROVED** [review]
Approved once the open comments are addressed.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-cli-pr-130-nmvb5 for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**